delphi-frontend/src/main/antlr3/au/com/integradev/delphi/antlr/Delphi.g

```diff
@@ -328,13 +330,19 @@ import org.apache.commons.lang3.StringUtils;
     return t;
   }
 
-  private Token combineLastNTokens(int count) {
+  private Token combineLastNTokens(int type, int count) {
     CommonToken firstToken = (CommonToken) input.LT(-count);
     CommonToken lastToken = (CommonToken) input.LT(-1);
+    lastToken.setType(type);
     lastToken.setStartIndex(firstToken.getStartIndex());
     return lastToken;
   }
 
+  private BinaryExpressionNodeImpl createBinaryExpression(Object operator) {
+    Token token = adaptor.getToken(operator);
+    return new BinaryExpressionNodeImpl(token);
+  }
```

Comment:
To make this a little more general-purpose, could we make this generic? Maybe something like `transformNode<T>`?

Reply:
No, not really.

You'd have to pass a `Class<T>` into the function and construct the object via reflection, which I imagine would be a quite a bit slower than what we're currently doing.

We probably don't need a general-purpose version anyway, as I'd rather eat a bowl of glass (without any milk) than pepper the grammar with something like `transformNode`. It's a heinous band-aid.

Reply:
Ugh, Java generics.